### PR TITLE
CMDCT-3072: [PDF] Initiatives Section

### PR DESCRIPTION
**Pull request by karla-vm**
### Description

Building out the PDF view of the State and Territory Specific Initiatives section, including each initiative's steps and their corresponding data. 

**NOTE:** This PR covers the State and Territory Initiatives and their individual steps.

co-authored with @britt-mo and @ntsummers1 ✨ 

### Related ticket(s)

CMDCT-3072

---
### How to test

- Log in as a state user
- Create a Work Plan
- Fill in the `Initiatives` dashboard steps
- Go to the `Review & Submit` page
- Click **Review PDF**

The last section _should_ match [Figma](https://www.figma.com/file/oOJMUZerbafZefLQQ33GwI/MFP_Playground?type=design&node-id=2413%3A99200&mode=design&t=7IZ4GQCF7T3ucW4t-1)

### Important updates

N/A

---
### Author checklist

- [ ] I have performed a self-review of my code
- [ ] I have added [thorough](https://shorturl.at/aejkF) tests, if necessary
- [ ] I have updated relevant documentation, if necessary
---

_convert to a different template: [test → val](?expand=1&template=test-to-val-deployment.md)_ | _[val → prod](?expand=1&template=val-to-prod-deployment.md)_

### CODE REVIEW

**Review by ajaitasaini — COMMENTED** [review]
**PDF**
Screenshot 2023-12-04 at 3 02 49 PM

**WEB**
Screenshot 2023-12-04 at 3 03 16 PM

I'm seeing this error icon in the pdf even though I filled out the initiative and it has a green check mark in web

**Comment by karla-vm** [comment]
> **PDF** Screenshot 2023-12-04 at 3 02 49 PM
> 
> **WEB** Screenshot 2023-12-04 at 3 03 16 PM
> 
> I'm seeing this error icon in the pdf even though I filled out the initiative and it has a green check mark in web

Fixed it!

**Review by ajaitasaini — APPROVED** [review]
created two initiatives, seeing the correct data and statusing for both! great job y'all

File: services/ui-src/src/components/export/ExportedModalOverlayReportSection.tsx [hunk 1]
```diff
@@ -63,49 +101,113 @@ export function renderStatusIcon(status: boolean) {
...
+                  return (
+                    <Box key={idx}>
+                      <ExportedOverlayModalReportSection
+                        section={section as OverlayModalPageShape}
+                        entity={entity}
+                        entityStep={step}
+                      />
+                    </Box>
+                  );
+                // TODO: Once we are tracking the close-out information step, we'll need to add it here
```
Comment (ntsummers1) [inline services/ui-src/src/components/export/ExportedModalOverlayReportSection.tsx:205]: Do we have a ticket to capture this to make sure we take care of it?

File: services/ui-src/src/components/export/ExportedSectionHeading.tsx [hunk 2]
```diff
@@ -52,7 +41,7 @@ const sx = {
   },
   heading: {
     margin: "1.5rem 0",
-    fontSize: "xl",
+    fontSize: "24px",
```
Comment (ntsummers1) [inline services/ui-src/src/components/export/ExportedSectionHeading.tsx:44]: Can
```suggestion
    fontSize: "2xl",
```

File: services/ui-src/src/components/export/ExportedOverlayModalReportSection.tsx [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+  stepName: {
+    fontSize: "18px",
+    paddingBottom: "0.75rem",
+  },
+  stepHint: {
+    fontSize: "16px",
+    fontWeight: "normal",
+    color: "#5B616B",
+  },
+  // TODO: delete this
```
Comment (ntsummers1) [inline services/ui-src/src/components/export/ExportedOverlayModalReportSection.tsx:94]: Do we still want to delete this?

File: services/ui-src/src/components/export/ExportedOverlayModalReportSection.tsx [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    fontSize: "md",
+    fontWeight: "bold",
+    color: "palette.gray_medium",
+  },
+  stepName: {
+    fontSize: "18px",
+    paddingBottom: "0.75rem",
+  },
+  stepHint: {
+    fontSize: "16px",
```
Comment (ntsummers1) [inline services/ui-src/src/components/export/ExportedOverlayModalReportSection.tsx:90]:
```suggestion
    fontSize: "md",
```

File: services/ui-src/src/components/export/ExportedOverlayModalReportSection.tsx [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+    marginTop: "0.5rem",
+  },
+  dashboardTitle: {
+    margin: "1rem auto 1.25rem",
+    fontSize: "md",
+    fontWeight: "bold",
+    color: "palette.gray_medium",
+  },
+  stepName: {
+    fontSize: "18px",
```
Comment (ntsummers1) [inline services/ui-src/src/components/export/ExportedOverlayModalReportSection.tsx:86]:
```suggestion
    fontSize: "lg",
```

File: services/ui-src/src/components/export/ExportedModalOverlayReportSection.tsx [hunk 6]
```diff
@@ -63,49 +101,113 @@ export function renderStatusIcon(status: boolean) {
...
+              <Td sx={sx.statusIcon}>
+                <EntityStatusIcon
+                  entity={entity}
+                  isPdf={true}
+                  entityStatus={getInitiativeStatus(report, entity)}
+                />
+              </Td>
+              <Td>
+                <Heading sx={sx.heading} as="h3">
+                  {`${idx + 1}. ${entity.initiative_name}` ?? "Not entered"}{" "}
```
Comment (ntsummers1) [inline services/ui-src/src/components/export/ExportedModalOverlayReportSection.tsx:164]:
```suggestion
                  {`${idx + 1}. ${entity.initiative_name}` ?? "Not entered"}{" "}
```

File: services/ui-src/src/components/export/ExportedEntityDetailsOverlaySection.tsx [hunk 7]
```diff
@@ -196,9 +169,14 @@ const sx = {
...
-    padding: "1rem 0 1rem 0",
-    fontWeight: "bold",
+  stepName: {
+    fontSize: "18px",
+    paddingBottom: "0.75rem",
+  },
+  stepHint: {
+    fontSize: "16px",
+    fontWeight: "normal",
+    color: "#5B616B",
```
Comment (ntsummers1) [inline services/ui-src/src/components/export/ExportedEntityDetailsOverlaySection.tsx:179]:
```suggestion
  stepName: {
    fontSize: "lg",
    paddingBottom: "0.75rem",
  },
  stepHint: {
    fontSize: "md",
    fontWeight: "normal",
    color: "#5B616B",
```

Comment (ntsummers1) [reply services/ui-src/src/components/export/ExportedEntityDetailsOverlaySection.tsx:179]: I'm starting to see the #5B616B crop up in a couple places. We should define it in the colors -> palette custom section in the theme.ts file

**Review by ntsummers1 — APPROVED** [review]
These are all nits so far. Feel free to tackle or not!

**Comment by ntsummers1** [comment]
Screenshot 2023-12-04 at 5 59 45 PM

Could use some gap between these 2 sections

**Review by ntsummers1 — CHANGES_REQUESTED** [review]
Overall, a few really tiny nits revolving around styling and what not. Very good work though! The PDF is looking strong.

I do have one request though before I approve this; I currently am seeing 43 errors in the console when loading the pdf. A lot of its related to key's being used multiple times. We should really clean this console up and not have the same keys when indexing things.

File: services/ui-src/src/components/export/ExportedModalOverlayReportSection.tsx (hunk 1 above)
Comment (britt-mo) [reply services/ui-src/src/components/export/ExportedModalOverlayReportSection.tsx:205]: @ntsummers1 I created card [CMDCT-3099](https://jiraent.cms.gov/browse/CMDCT-3099) for this

File: services/ui-src/src/components/export/ExportedOverlayModalReportSection.tsx (hunk 3 above)
Comment (britt-mo) [reply services/ui-src/src/components/export/ExportedOverlayModalReportSection.tsx:94]: deleted, wasn't being used

**Review by ntsummers1 — APPROVED** [review]

**Pull request merged**